data/postRouter.js

```diff
@@ -0,0 +1,73 @@
+const router = require('express').Router();
+
+const Posts = require('../data/db.js');
+
+router.post('/', (req, res) => {
+    const { title, contents } = req.body;
+    if(!title || !contents){
+        res.status(400).json({ message: 'Please provide title and contents for the post.' })
+    }
+    Posts.insert({ title, contents }).then( newPost => {
+        res.status(201).json(newPost);
+    }).catch(error => {
+        res.status(500).json({ message: 'There was an error while saving the post to the database.' })
+    })
+})
+
+router.get('/', (req, res) => {
+    Posts.find().then( post => {
+        res.status(200).json(post)
+    }).catch(error => {
+        res.status(500).json({ message: 'The posts information could not be retrieved.' })
+    })
+})
+
+router.get('/:id', (req, res) => {
+    const id = req.params.id;
+    Posts.findById(id).then(post => {
+        if(id === 0){ //logic on this line isn't working
```

Comment:
if(!post). Id is coming from the params, you will always have an id because this is the `/:id` route